dissect/target/plugins/filesystem/ntfs/mft.py

```diff
@@ -53,21 +72,57 @@
     ],
 )
 
+FilesystemFilenameRecord = TargetRecordDescriptor(
+    "filesystem/ntfs/mft/filename",
+    [
+        ("datetime", "ts"),
+        ("string", "ts_type"),
+        ("uint32", "filename_index"),
+        ("uint32", "segment"),
+        ("uri", "path"),
+        ("string", "owner"),
+        ("filesize", "filesize"),
+        ("boolean", "resident"),
+        ("boolean", "inuse"),
+        ("boolean", "ads"),
+        ("string", "volume_uuid"),
+    ],
+)
+
+
+RECORD_TYPES = {
+    InformationType.STANDARD_INFORMATION: FilesystemStdRecord,
+    InformationType.FILE_INFORMATION: FilesystemFilenameRecord,
+}
+
+
+COMPACT_RECORD_TYPES = {
+    InformationType.STANDARD_INFORMATION: FilesystemStdRecordCompacted,
+    InformationType.FILE_INFORMATION: FilesystemFilenameRecordCompacted,
+}
+
 
 class MftPlugin(Plugin):
     def check_compatible(self):
         ntfs_filesystems = [fs for fs in self.target.filesystems if fs.__fstype__ == "ntfs"]
         return len(ntfs_filesystems) > 0
 
-    @export(record=[FilesystemStdRecord, FilesystemFilenameRecord])
-    def mft(self):
+    @export(record=[FilesystemStdRecordCompacted, FilesystemFilenameRecordCompacted])
+    @arg("--compact", type=bool)
```

Comment:
```suggestion
    @arg("--compact", action="store_true", type=bool)
```
Or can `argparse` infer that from the type?